internal/hashing/reader.go

```diff
@@ -5,25 +5,51 @@ import (
 	"io"
 )
 
-// Reader hashes all data read from the underlying reader.
-type Reader struct {
+// ReadSumer hashes all data read from the underlying reader.
+type ReadSumer interface {
+	Read(p []byte) (int, error)
+	// Sum returns the hash of the data read so far.
+	Sum(d []byte) []byte
+}
+
+type reader struct {
 	r io.Reader
 	h hash.Hash
 }
 
-// NewReader returns a new Reader that uses the hash h.
-func NewReader(r io.Reader, h hash.Hash) *Reader {
-	return &Reader{
+type readWriterTo struct {
+	reader
+	writerTo io.WriterTo
```

Comment:
There is no need to name the `io.WriterTo` here..

Reply:
I'd prefer to keep the name, as the goal here is not to directly expose the `io.WriterTo` object but instead to always wrap it.